### PR TITLE
[APPSRE-3325]Introduce code coverage in tox configs

### PR DESCRIPTION
**Pull request by Piojo**
To get the tests ball rolling. We seem to be using tox for testing, so
I adapt its configuration to generate reports.

The minimum required coverage is 19%. We currently have 19.07%
coverage.

I also disable the `test_make` file, which doesn't work at all on my
system, even without any local changes to my git repository.

### CODE REVIEW

File: reconcile/test/test_make.py [hunk 1]
```diff
@@ -17,6 +19,9 @@ def has_uncommited_changes():
 
 class TestMake:
     @staticmethod
+    @pytest.mark.skip(
+        "Broken - uncommitted changes are normal during development"
+        )
```
Comment (maorfr) [inline reconcile/test/test_make.py:24]: i'd advise not to disable this. this test is here due to past issues with folks not generating the openshift templates using `make generate`.

**Comment by Piojo** [comment]
> i'd advise not to disable this. this test is here due to past issues
> with folks not generating the openshift templates using `make
> generate`.

I was sure there was a good reason for this test. But it doesn't even
pass on master.

If we want to run it as part of a pipeline, we may use a `skipIf` gate
instead, but then, how do you recommend to identify that we are running
in a CI environment?

**Comment by maorfr** [comment]
perhaps there is an environment variable we can use that exists when running in jenkins?

File: tox.ini [hunk 2]
```diff
@@ -5,10 +5,23 @@ envlist = py36
 commands =
     flake8 reconcile tools e2e_tests
     pylint -j0 reconcile tools e2e_tests
-    pytest
+    pytest --cov=reconcile
```
Comment (apahim) [inline tox.ini:8]: let's also add `--cov-report=term-missing`. That's shows the gaps.

**Comment by Piojo** [comment]
New version undergoing CI. Will `HUDSON_HOME` stay even as pipelines move to Tekton?

File: tox.ini (hunk 2 above)
Comment (Piojo) [reply tox.ini]: Fixed!

**Comment by maorfr** [comment]
i don't think we will find something that will appear both in jenkins and in tekton. we can control what will go into tekton, and that is a long way ahead anyway, so i would separate the two concerns.

HUDSON_HONE is jenkins specific and so is JOB_*. but than again, probably everything else. there are other CI systems which expose something like `CI=true`, but we don't have it here.

i think we have 2 options:
1. choose something generic, like GIT_BRANCH for example (even though it will not be compatible with tekton)
2. enhance all our jobs to have something generic like `CI=true` (we can implement the same in tekton for future compatibility)

**Comment by Piojo** [comment]
Ok. These tests are going to run in jenkins for the time being (what are the plans for tekton and ci-ext?), so I propose we move forward as it is and revisit when we change the CI service for this repo.

**Review by maorfr — APPROVED** [review]
LGTM

**Comment by maorfr** [comment]
these are the current plans for migrating this sort of job: https://issues.redhat.com/browse/APPSRE-3184

**Pull request merged**